### ISSUE TITLE
Error parsing wp-cli.yml

### ISSUE THREAD

**Issue by ParhamG** [wp-cli/wp-cli#5257]
I am trying to define aliases in wp-cli.yml file:

```
@local:
     ssh: vagrant@www.site1.test
     path: /srv/www/site1.test/htdocs/
     url: http://www.site1.test
```
but then running a simple `wp @local core version` gives me the following error after entering the password for the remote vagrant box:
```
PHP Warning:  Invalid argument supplied for foreach() in phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Configurator.php on line 98
Error: Alias '@local' not found.
```

I have found out if I only have one attribute (e.g. `ssh`) in the yml file and leave out the others:
```
@local:
     ssh: vagrant@www.site1.test
```
then running the following command works just fine:
`wp @local core version --path=/srv/www/site1.test/htdocs/ --url=http://www.site1.test` with the output `5.2.2`

Looking at the source code for `Configurator.php` it should only get to line 98 if the env variable `WP_CLI_RUNTIME_ALIAS` is defined which is not in my case (neither on local nor the remote box)

-------------
Here is another example, with a simple yml file:
```
ssh: vagrant@www.site1.test
path: /srv/www/site1.test/htdocs/
url: http://www.site1.test
```
running the command `wp core version` fails with the following error after asking for the password of the vagrant box:
```
Error: This does not seem to be a WordPress install.
Pass --path=`path/to/wordpress` or run `wp core download`.
```
which means it has pulled the `ssh` attribute from the yml file but not the other attributes.

----------------
Here is the wp cli info for the host machine:
```
OS:     Windows NT 10.0 build 18362 (Windows 10) AMD64
Shell:  C:\WINDOWS\system32\cmd.exe
PHP binary:     C:\Program Files\PHP\php.exe
PHP version:    7.1.12
php.ini used:   C:\Program Files\PHP\php.ini
WP-CLI root dir:        phar://wp-cli.phar/vendor/wp-cli/wp-cli
WP-CLI vendor dir:      phar://wp-cli.phar/vendor
WP_CLI phar path:       C:\WordPress\Tools\wp-cli
WP-CLI packages dir:
WP-CLI global config:
WP-CLI project config:  C:\WordPress\Tools\wp-cli\wp-cli.yml
WP-CLI version: 2.2.0
```

Here is the wp cli info for the remote vagrant box:
```
OS:     Linux 4.15.0-45-generic #48-Ubuntu SMP Tue Jan 29 16:28:13 UTC 2019 x86_64
Shell:  /bin/bash
PHP binary:     /usr/bin/php7.2
PHP version:    7.2.19-1+ubuntu18.04.1+deb.sury.org+1
php.ini used:   /etc/php/7.2/cli/php.ini
WP-CLI root dir:        phar://wp-cli.phar/vendor/wp-cli/wp-cli
WP-CLI vendor dir:      phar://wp-cli.phar/vendor
WP_CLI phar path:       /srv/www/site1.test/htdocs
WP-CLI packages dir:
WP-CLI global config:
WP-CLI project config:  /srv/www/site1.test/wp-cli.yml
WP-CLI version: 2.3.0-alpha-b0be5c1
```

**Comment by schlessera** [comment]
@ParhamG I cannot reproduce this with a regular alias to an SSH host. I assume it is specific to the special-case `vagrant` handling, but can't currently test that.

We recetnly fixed some inconsistencies with this special protocol here: https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli-nightly.phar

Let me try to get this into a fresh nightly. I'll ping you once this is done so you can test whether this fixes the issue.

**Comment by ParhamG** [comment]
I think I figured this out, I was following the instructions and samples on the [Config](https://make.wordpress.org/cli/handbook/config/) page but it appears that according to the instructions on the page [Running Commands Remotely](https://make.wordpress.org/cli/handbook/running-commands-remotely/) for SSH the path must be added as a suffix to the `ssh` option instead of a separate `path` option. So this worked:

```
@sandbox:
  ssh: parhamg@sandbox.site.com/var/www/vhosts/sandbox.wordpress.site.com/httpdocs
```

**Comment by schlessera** [comment]
Yes, indeed, that is the correct syntax for the SSH setting.

However, it seems the error you're getting when misconfiguring SSH is very misleading. I've opened a new ticket to make this clearer: https://github.com/wp-cli/wp-cli/issues/5279

Glad you got this resolved.
